storage/NRF52FlashStorage.cpp

```diff
@@ -0,0 +1,287 @@
+/**
+ ******************************************************************************
+ * @file    FlashStorage.cpp
+ * @author  Waldemar Gruenwald
+ * @version V1.0.0
+ * @date    23 October 2017
+ * @brief   key-storage class implementation
+ *
+ * @update 	V1.0.1
+ * 			08 August 2017
+ * 			error handling and debug
+ ******************************************************************************
+ *
+ * Copyright 2016 ubirch GmbH (https://ubirch.com)
+ *
+ * ```
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ```
+ */
+
+/**
+ *  For more information about the key-storage,
+ *  see documentation for mbed fstorage
+ */
+#include "FlashStorage.h"
+#include <nrf_soc.h>
+#include <BLE.h>
+#include <EventQueue.h>
+#include "NRF52FlashStorage.h"
+
+#include "mbed.h"
+
+extern "C" {
+#include <softdevice_handler.h>
+
+}
+
+
+//#define PRINTF(...)
+#define PRINTF printf
+
+// TODO, maybe separate the storage into two different storages, for secure and not secure data
+
+static Thread *sdEventThread;//(osPriorityNormal);
```

Comment:
it may make sense to put these into the Flashstorage class as member fields since we are creating and destroying them on-demand anyway.